Main svc

OsipovKonstantin opened the pull request.

OsipovKonstantin committed:
feat: добавление для основного сервиса general-service схемы БД, классов-dto, классов-сущностей

OsipovKonstantin committed:
feat: добавление для основного сервиса general-service схемы БД, классов-dto, классов-сущностей

OsipovKonstantin committed:
feat: добавление основной функциональности general-service. Схема БД, сущности, дто, мапперы, эндпоинты, сервисы, репозитории, всё согласно спецификации Swagger. Налаживание связи между модулями general-service и stats-service с помощью stats-client'а. Описание ошибок в более подробном формате ApiError (возвращаемый класс методов ErrorHandler)

OsipovKonstantin committed:
fix: устранение неполадок для прохождения тестов github

avfyodorov reviewed (commented):
Добрый день!

Замечаний нет.
Работа принята.

OsipovKonstantin committed:
refactor and fix:
добавление логирования (2 строки в application.properties);
добавление валидации минимального значения @Min для полей from и size в запросах;
@Min и @Max для lat и long - полей класса Location;
помещение классов с Location в отдельный пакет, так как это самостоятельная сущность;
добавление ограничения @PositiveOrZero для поля participantLimit;
добавление метода existsById в сервисы как более удобная альтернатива и дополнение к findById;
добавление обработки остальных ошибок в ErrorHandler, для них будет сгенерирована ошибка 500 Internal Server Error;
группировка ошибок в меньшее кол-во - NotFoundException, ValidationException и ConflictException и задание более простого описания ошибок в ErrorHandler;
извлечение сообщения об ошибке для MethodArgumentNotValidException в ErrorHandler (e.getBindingResult().getFieldErrors()...)

OsipovKonstantin commented:
> Добрый день!
> 
> Замечаний нет. Работа принята.

Спасибо вам!
